### PR TITLE
Fix `intersect` with frozen objects

### PR DESCRIPTION
**Pull request by compulim**
## Summary

Fixed `intersect` should not modify object inline as it could be frozen, instead, it should modify a clone of the object/array.

## Background

Sometimes, for performance reason, instead of returning a new empty array, we would return a flywheel object instead. That flywheel object is frozen to prevent cross pollution.

```ts
const EMPTY_ARRAY = Object.freeze([]);

const someSchema = pipe(
  array(string()),
  readonly(),
  transform(value => value.length ? value : EMPTY_ARRAY)
);
```

When combining with `intersect`, the above schema would fail. This is because `intersect` is trying to modify the output of the schema, which is frozen/sealed.

IMO, most functions should not inline-modify input unless they own the object.

---
## Summary by cubic
Fixes `intersect` to work with frozen objects and arrays by merging into shallow clones instead of mutating inputs. Prevents runtime errors when schemas return frozen or readonly values and keeps outputs immutable.

- **Bug Fixes**
  - Merge into clones in `_merge` (`nextValue = { ...value1 }`, `nextValue = [...value1]`) and return the clone.
  - Add tests for frozen objects (including empty), frozen arrays, and `intersect` with `readonly()` + `transform` that freezes outputs.
  - Document the fix in `CHANGELOG.md`.

<sup>Written for commit 6ae72aca90b8a7c810a240a0a81def8d3a570f6a. Summary will update on new commits.</sup>

### CODE REVIEW

File: library/src/schemas/intersect/utils/_merge/_merge.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ export function _merge(value1: unknown, value2: unknown): MergeDataset {
       value1.constructor === Object &&
       value2.constructor === Object
     ) {
+      const nextValue = { ...value1 };
+
       // Deeply merge entries of `value2` into `value1`
```
[the remark this answers is not in the document]
Comment (compulim) [reply library/src/schemas/intersect/utils/_merge/_merge.ts]: Fixed.

**Comment by fabian-hiller** [comment]
Valibot v1.4.0 is available
